data/serializer/src/main/java/tech/pegasys/teku/api/schema/BeaconBlockBody.java

```diff
@@ -25,25 +32,75 @@
 import tech.pegasys.teku.spec.datastructures.blocks.blockbody.BeaconBlockBodySchema;
 
 @SuppressWarnings("JavaCase")
 public class BeaconBlockBody {
   @Schema(type = "string", format = "byte", description = DESCRIPTION_BYTES96)
-  public final BLSSignature randao_reveal;
+  public BLSSignature randao_reveal;
 
-  public final Eth1Data eth1_data;
+  public Eth1Data eth1_data;
 
   @Schema(type = "string", format = "byte", description = DESCRIPTION_BYTES32)
-  public final Bytes32 graffiti;
+  public Bytes32 graffiti;
 
-  public final List<ProposerSlashing> proposer_slashings;
-  public final List<AttesterSlashing> attester_slashings;
-  public final List<Attestation> attestations;
-  public final List<Deposit> deposits;
-  public final List<SignedVoluntaryExit> voluntary_exits;
+  public List<ProposerSlashing> proposer_slashings;
+  public List<AttesterSlashing> attester_slashings;
+  public List<Attestation> attestations;
+  public List<Deposit> deposits;
+  public List<SignedVoluntaryExit> voluntary_exits;
+
+  public static final DeserializableTypeDefinition<BeaconBlockBody> BEACON_BLOCK_BODY_TYPE =
+      DeserializableTypeDefinition.object(BeaconBlockBody.class)
+          .initializer(BeaconBlockBody::new)
+          .withField(
+              "randao_reveal",
+              BLS_SIGNATURE_TYPE,
+              BeaconBlockBody::getRandaoReveal,
+              BeaconBlockBody::setRandaoReveal)
+          .withField(
+              "eth1_data",
+              ETH_1_DATA_TYPE,
+              BeaconBlockBody::getEth1Data,
+              BeaconBlockBody::setEth1Data)
+          .withField(
+              "graffiti",
+              CoreTypes.BYTES32_TYPE,
+              BeaconBlockBody::getGraffiti,
+              BeaconBlockBody::setGraffiti)
+          .withField(
+              "proposer_slashings",
+              new DeserializableListTypeDefinition<>(PROPOSER_SLASHING_TYPE),
+              BeaconBlockBody::getProposerSlashings,
+              BeaconBlockBody::setProposerSlashings)
+          .withField(
+              "attester_slashings",
+              new DeserializableListTypeDefinition<>(ATTESTER_SLASHING_TYPE),
+              BeaconBlockBody::getAttesterSlashings,
+              BeaconBlockBody::setAttesterSlashings)
+          .withField(
+              "attestations",
+              new DeserializableListTypeDefinition<>(ATTESTATION_TYPE),
+              BeaconBlockBody::getAttestations,
+              BeaconBlockBody::setAttestations)
+          .withField(
+              "deposits",
+              new DeserializableListTypeDefinition<>(DEPOSIT_TYPE),
+              BeaconBlockBody::getDeposits,
+              BeaconBlockBody::setDeposits)
+          .withField(
+              "voluntary_exits",
+              new DeserializableListTypeDefinition<>(SIGNED_VOLUNTARY_EXIT_TYPE),
+              BeaconBlockBody::getVoluntaryExits,
+              BeaconBlockBody::setVoluntaryExits)
+          .build();
```

Comment:
we should be able to get object type defs from any SSZ object?

also they definitely shouldn't be in 'schema' - the 'schema' objects are the non typedef...